Rename HighlightRangeData to HighlightRange as part of rename work to clarify classes around highlights.

megangardner opened the pull request:
#### 6920239d3ba0a14c48ca903c33218e4f4dafe6b1
<pre>
Rename HighlightRangeData to HighlightRange as part of rename work to clarify classes around highlights.
<a href="https://bugs.webkit.org/show_bug.cgi?id=261594">https://bugs.webkit.org/show_bug.cgi?id=261594</a>
rdar://115542469

Reviewed by Tim Horton.

Several classes associated with Highlights have a very confusing naming
structure. This is the first rename of several to help clarify
what these classes are and their relationship to each other.

* Source/WebCore/Modules/highlight/Highlight.cpp:
(WebCore::Highlight::Highlight):
(WebCore::Highlight::initializeSetLike):
(WebCore::Highlight::removeFromSetLike):
(WebCore::Highlight::clearFromSetLike):
(WebCore::Highlight::addToSetLike):
(WebCore::Highlight::repaint):
* Source/WebCore/Modules/highlight/Highlight.h:
(WebCore::HighlightRange::create):
(WebCore::HighlightRange::HighlightRange):
(WebCore::Highlight::highlightRanges const):
(WebCore::HighlightRangeData::create): Deleted.
(WebCore::HighlightRangeData::range const): Deleted.
(WebCore::HighlightRangeData::startPosition const): Deleted.
(WebCore::HighlightRangeData::setStartPosition): Deleted.
(WebCore::HighlightRangeData::endPosition const): Deleted.
(WebCore::HighlightRangeData::setEndPosition): Deleted.
(WebCore::HighlightRangeData::HighlightRangeData): Deleted.
(WebCore::Highlight::rangesData const): Deleted.
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::collectRangeDataFromRegister):
(WebCore::Document::updateHighlightPositions):
* Source/WebCore/dom/Document.h:
* Source/WebCore/rendering/HighlightData.cpp:
(WebCore::HighlightData::setRenderRange):
* Source/WebCore/rendering/HighlightData.h:
* Source/WebCore/rendering/MarkedText.cpp:
(WebCore::MarkedText::collectForHighlights):
* Source/WebCore/rendering/RenderReplaced.cpp:
(WebCore::RenderReplaced::calculateHighlightColor const):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::numberOfAppHighlights):

Canonical link: <a href="https://commits.webkit.org/268022@main">https://commits.webkit.org/268022@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0332baf986a18b596ab41ee4d50d56ec911e85bd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18383 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18721 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/19314 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/20232 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17211 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/18579 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/22027 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/18870 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/19123 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18607 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18803 "1 failures") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16012 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21111 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16026 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/16769 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/23249 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/17046 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/16939 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/21136 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17511 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/14856 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16593 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4380 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/20960 "Built successfully") | | | 
| | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17350 "Failed to compile WebKit") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebCore/Modules/highlight/Highlight.cpp:
```diff
@@ -55,52 +55,54 @@ Ref<Highlight> Highlight::create(FixedVector<std::reference_wrapper<WebCore::Abs
 bool Highlight::removeFromSetLike(const AbstractRange& range)
 {
-    return m_rangesData.removeFirstMatching([&range](const Ref<HighlightRangeData>& current) {
+    return m_highlightRanges.removeFirstMatching([&range](const Ref<HighlightRange>& current) {
```
Optional nit: The template arg for `Ref` probably doesn't need specifying, but that stretches the scope of this PR too far... maybe change that in a forthcoming refactor PR 😛

megangardner committed:
Rename HighlightRangeData to HighlightRange as part of rename work to clarify classes around highlights.
https://bugs.webkit.org/show_bug.cgi?id=261594
rdar://115542469

Reviewed by Tim Horton.

Several classes associated with Highlights have a very confusing naming
structure. This is the first rename of several to help clarify
what these classes are and their relationship to each other.

* Source/WebCore/Modules/highlight/Highlight.cpp:
(WebCore::Highlight::Highlight):
(WebCore::Highlight::initializeSetLike):
(WebCore::Highlight::removeFromSetLike):
(WebCore::Highlight::clearFromSetLike):
(WebCore::Highlight::addToSetLike):
(WebCore::Highlight::repaint):
* Source/WebCore/Modules/highlight/Highlight.h:
(WebCore::HighlightRange::create):
(WebCore::HighlightRange::HighlightRange):
(WebCore::Highlight::highlightRanges const):
(WebCore::HighlightRangeData::create): Deleted.
(WebCore::HighlightRangeData::range const): Deleted.
(WebCore::HighlightRangeData::startPosition const): Deleted.
(WebCore::HighlightRangeData::setStartPosition): Deleted.
(WebCore::HighlightRangeData::endPosition const): Deleted.
(WebCore::HighlightRangeData::setEndPosition): Deleted.
(WebCore::HighlightRangeData::HighlightRangeData): Deleted.
(WebCore::Highlight::rangesData const): Deleted.
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::collectRangeDataFromRegister):
(WebCore::Document::updateHighlightPositions):
* Source/WebCore/dom/Document.h:
* Source/WebCore/rendering/HighlightData.cpp:
(WebCore::HighlightData::setRenderRange):
* Source/WebCore/rendering/HighlightData.h:
* Source/WebCore/rendering/MarkedText.cpp:
(WebCore::MarkedText::collectForHighlights):
* Source/WebCore/rendering/RenderReplaced.cpp:
(WebCore::RenderReplaced::calculateHighlightColor const):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::numberOfAppHighlights):

Canonical link: https://commits.webkit.org/268022@main
